### PR TITLE
Fixes deep migration rates in OOA model

### LINKED ISSUE #981: Fix OOA example
As described here: https://github.com/popsim-consortium/stdpopsim/pull/496 the example [in the docs](https://msprime.readthedocs.io/en/latest/tutorial.html#id1) is slightly wrong - this should be fixed and the subtle reason it was wrong should be explained.

### PR DESCRIPTION
**Pull request by apragsdale**
Fixes migration matrix error in deep past and includes a note block describing the error. Also includes a schematic of this demographic model.

Closes #981.

### CODE REVIEW

File: docs/tutorial.rst [hunk 1]
```diff
@@ -706,6 +716,7 @@ with most recent events first.
             # Population B merges into YRI at T_B
             msprime.MassMigration(
                 time=T_B, source=1, destination=0, proportion=1.0),
+            msprime.MigrationRateChange(time=T_B, rate=0,
```
Comment (petrelharp) [inline docs/tutorial.rst:719]: aren't you missing a `)` here?

**Comment by petrelharp** [comment]
Not for this PR, but (as you've probably thought about) a useful addition to the DemographyDebugger printing-out is an indication of which populations might have lineages in them, using #969.

File: docs/tutorial.rst (hunk 1 above)
Comment (apragsdale) [reply docs/tutorial.rst:719]: Yes! will fix.

**Comment by petrelharp** [comment]
Looks like the output from `dd.print_history()` needs to be updated also: here's what I get:
```
Model =  hudson(reference_size=1)
=============================
Epoch: 0 -- 848.0 generations
=============================
     start     end      growth_rate |     0        1        2    
   -------- --------       -------- | -------- -------- -------- 
0 |1.23e+04 1.23e+04              0 |     0      3e-05   1.9e-05 
1 |2.97e+04   1e+03           0.004 |   3e-05      0     9.6e-05 
2 |5.41e+04    510           0.0055 |  1.9e-05  9.6e-05     0    

Events @ generation 848.0
   - Mass migration: Lineages moved with probability 1.0 backwards in time with source 2 & dest 1
                     (equivalent to migration from 1 to 2 forwards in time)
   - Migration rate change to 0 everywhere
   - Migration rate change for (0, 1) to 0.00025
   - Migration rate change for (1, 0) to 0.00025
   - Population parameter change for 1: initial_size -> 2100 growth_rate -> 0 
==================================
Epoch: 848.0 -- 5600.0 generations
==================================
     start     end      growth_rate |     0        1        2    
   -------- --------       -------- | -------- -------- -------- 
0 |1.23e+04 1.23e+04              0 |     0     0.00025     0    
1 | 2.1e+03  2.1e+03              0 |  0.00025     0        0    
2 |   510   2.27e-09         0.0055 |     0        0        0    

Events @ generation 5600.0
   - Mass migration: Lineages moved with probability 1.0 backwards in time with source 1 & dest 0
                     (equivalent to migration from 0 to 1 forwards in time)
   - Migration rate change to 0 everywhere
===================================
Epoch: 5600.0 -- 8800.0 generations
===================================
     start     end      growth_rate |     0        1        2    
   -------- --------       -------- | -------- -------- -------- 
0 |1.23e+04 1.23e+04              0 |     0        0        0    
1 | 2.1e+03  2.1e+03              0 |     0        0        0    
2 |2.27e-09 5.17e-17         0.0055 |     0        0        0    

Events @ generation 8800.0
   - Population parameter change for 0: initial_size -> 7300 
================================
Epoch: 8800.0 -- inf generations
================================
     start     end      growth_rate |     0        1        2    
   -------- --------       -------- | -------- -------- -------- 
0 | 7.3e+03  7.3e+03              0 |     0        0        0    
1 | 2.1e+03  2.1e+03              0 |     0        0        0    
2 |5.17e-17     0            0.0055 |     0        0        0    
```

**Comment by apragsdale** [comment]
> Looks like the output from `dd.print_history()` needs to be updated also

It's not jumping out to me.. What's the issue with the printed history here?

**Comment by jeromekelleher** [comment]
LGTM, thanks @apragsdale. Can you merge when you're happy please @petrelharp?

**Comment by petrelharp** [comment]
It was still printing the (wrong) ancient migration - I've fixed it and made you a PR against your branch; lmk if there's issues.

**Comment by petrelharp** [comment]
Whoops - I had a bit of a mixup with branches, but it it's all fixed up now, I think?

**Comment by petrelharp** [comment]
I think this is good! - could you squash these commits down?

**Comment by apragsdale** [comment]
I'm having some git issues... trying to pull your merged changes locally so I can squash the commits and repush, but can't seem to get it to work. Also, did your PR also accidentally include changes to the lineage probability functions? Because for some reason those changes are now also on this OOA_fix branch.

**Comment by apragsdale** [comment]
Ok, I just ended up copy-pasting that fix to the demography debugger output (I've been seeing weird mashups of different branches). If this looks good and squashed, I think it's ready to merge.

**Comment by petrelharp** [comment]
> Also, did your PR also accidentally include changes to the lineage probability functions?

Oh dear, quite possibly. I got myself in quite a git tangle this morning. Sorry about that.

**Pull request merged**